sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_client_manager.py

```diff
@@ -303,37 +309,73 @@ def __init__(
             replica_discovery_enabled,
             min_backoff_sec,
             max_backoff_sec,
+            load_balancing_enabled,
             **kwargs
         )
         self._original_connection_string = connection_string
         self._credential = credential
+        self._replica_clients: List[_ConfigurationClientWrapper] = []
+        self._active_clients: List[_ConfigurationClientWrapper] = []
 
         if connection_string and endpoint:
-            self._replica_clients.append(
-                _ConfigurationClientWrapper.from_connection_string(
+            self._original_client = _ConfigurationClientWrapper.from_connection_string(
                     endpoint, connection_string, user_agent, retry_total, retry_backoff_max, **self._args
                 )
-            )
-            self._setup_failover_endpoints()
-            return
-        if endpoint and credential:
-            self._replica_clients.append(
-                _ConfigurationClientWrapper.from_credential(
+            self._replica_clients.append(self._original_client)
+        elif endpoint and credential:
+            self._original_client =  _ConfigurationClientWrapper.from_credential(
                     endpoint, credential, user_agent, retry_total, retry_backoff_max, **self._args
                 )
-            )
-            self._setup_failover_endpoints()
+            self._replica_clients.append(self._original_client)
+        else:
+            raise ValueError("Please pass either endpoint and credential, or a connection string with a value.")
+
+    def get_next_client(self) -> Optional[_ConfigurationClientWrapper]:
+        """
+        Get the next client to be used for the request. find_active_clients needs be called before this method or None
+        client will be returned. If load_balancing_enabled isn't enabled the first active client will be returned,
+        which is the provided endpoint unless it has failed. If load_balancing_enabled is enabled the next client in
+        the list will be returned.
```

Comment:
```suggestion
        Get the next active client to be used for the request. if `find_active_clients` has never been invoked, this method returns None.
```

The logic around load balancing to determine active client is defined in find_active_clients method. This method just returns the first active client always.

Reply:
Same comment for async